.github/workflows/ci.yml

```diff
@@ -51,3 +52,18 @@ jobs:
       - run: cabal build
       - run: cabal test
       - run: ./script/haddock
+
+  release:
+    needs: build
+    if: github.event_name == 'release'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Generate package ready for Hackage distribution
+        run: cabal sdist
+
+      - name: Upload package to Hackage
+        run: |
+          cabal upload --publish \
+            --username '${{ secrets.HACKAGE_USERNAME }}' \
+            --password '${{ secrets.HACKAGE_PASSWORD }}' \
+            dist-newstyle/sdist/hapistrano-${{ github.event.release.tag_name }}.tar.gz
```

Comment:
@DavidMazarro I think you can get rid of `github.event.release.tag_name` here in favor of `*`, something like this:

`dist-newstyle/sdist/hapistrano-*.tar.gz`

Reply:
We can do this, it's true that it will be simpler. I was a bit concerned about compatibility because glob expressions don't behave consistently across different shells; but I guess if it's always going to be performed under `ubuntu-latest` without any weird tweaks that shouldn't be an issue.